### ISSUE TITLE
KiPiSH: backend, diffable FullState abstraction

### ISSUE THREAD

**Issue by jayunit100** [kubernetes-sigs/kpng#215]
Lets build a diff tool that backends can reuse that replaces serviceChangeTracker ***** FORVER **** 

@hanamantagoudvk 

# DETAILS

## Problem:

IPVS backends dont use serviceChangeTracker but need a diff solution for 

```
AddSessionAffinity -- ..
InternalTrafficPolicy
ExternalTrafficPolicy
TopologyAwareHints
```

## Solution:

talking w/ @hanamantagoudvk @mcluseau @knabben @dougsland 

- kpng backends get SetService/SetEndpoint for every single K8s delta
- its cheap and easy to cache these in Go locally
- its cheap to just calculate a "diff" on the fly
- we thus dont probably need the dang serviceChangeTracker anymore
- that said we use it for iptables, we can eventually replace it with the byproduct of this issue which is 

A client side cache-diff-tool `KCS`

# Nice to have

we should throw a burial party for SCT when this is completed

**Comment by jayunit100** [comment]
This is a good newcomer issue for the india team i think - so once the library is done maybe we can add unit tests after the fact as a follow on `good-first-issue` @hanamantagoudvk please decompose if you can so we can leverage this to grow the overall group of folks that know how to do this
